Divine/CLI/CommandLineLogger.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using Divine.Enums;
+
+namespace Divine.CLI
+{
+    internal class CommandLineLogger
+    {
+        private static readonly LogLevel optionLogLevel = CommandLineActions.LogLevel;
+
+        public static void LogInfo(string message) => Log(LogLevel.INFO, message);
+        public static void LogWarn(string message) => Log(LogLevel.WARN, message);
+        public static void LogError(string message) => Log(LogLevel.ERROR, message);
+        public static void LogFatal(string message) => Log(LogLevel.FATAL, message);
+        public static void LogDebug(string message) => Log(LogLevel.DEBUG, message);
+
+        private static void Log(LogLevel loglevel, string message)
+        {
+            if (optionLogLevel == LogLevel.SILENT) return;
+
+            switch (loglevel)
+            {
+                case LogLevel.INFO:

```

Comment:
Small nitpick: It'd probably be more readable if the log levels were ordered by their importance in this switch and other places, i.e. DEBUG -> INFO -> WARN -> etc., instead of being in a seemingly random order.